### PR TITLE
Begin work to get phytool ready for Fedora inclusion.

### PR DESCRIPTION
**Pull request by smooge**
- Fedora and other distros add a lot of flags to uncover possible problems with code. Added in an error check on asprintf as it could fail and cause problems.

### CODE REVIEW

**Comment by wkz** [comment]
I'm not opposed to merging these error checks, but the formatting of this change needs some work.

Please set your editor to use linux kernel style indentation, and update this PR with the results.

**Comment by smooge** [comment]
omg. I just looked at that and realized my mistake with all that white space. I am fixing this.

**Comment by smooge** [comment]
I have some other proposed changes which came up in the package review but want to make sure I am following how you would like them. Is it better to make more PR's or first make issues and then PR's

**Comment by smooge** [comment]
And I have made a bodge of this PR again by adding in the Makefile changes I had found. My apologies

**Comment by wkz** [comment]
No worries  :smile: 

Please create PRs directly, there's no need to have an issue for it as well.

**Comment by smooge** [comment]
I have tried to clean out the problems but it is a lot of differnet ones now combined. I can open a new PR with this all squashed or just live with this. The Makefile changes for dealing with md5sum and a missing bin will be a different PR

File: phytool.c [hunk 1]
```diff
@@ -275,13 +275,19 @@ static int mv6tool_parse_loc_if(char *dev, char *addr, char *reg,
 
 	strncpy(loc->ifnam, dev, IFNAMSIZ - 1);
 
-	asprintf(&path, "/sys/class/net/%s/phys_switch_id", dev);
+	err = asprintf(&path, "/sys/class/net/%s/phys_switch_id", dev);
+	if (err == -1 )
```
Comment (wkz) [inline phytool.c]:
```suggestion
	if (err == -1)
```

File: phytool.c [hunk 2]
```diff
@@ -275,13 +275,19 @@ static int mv6tool_parse_loc_if(char *dev, char *addr, char *reg,
...
+		return -ENOMEM;
+
 	err = sysfs_readu(path, &phy_port);
 	free(path);
 	if (err)
 		return -ENOSYS;
 
-	asprintf(&path, "/sys/class/net/%s/phys_port_id", dev);
+	err =asprintf(&path, "/sys/class/net/%s/phys_port_id", dev);
```
Comment (wkz) [inline phytool.c]:
```suggestion
	err = asprintf(&path, "/sys/class/net/%s/phys_port_id", dev);
```

File: phytool.c [hunk 3]
```diff
@@ -275,13 +275,19 @@ static int mv6tool_parse_loc_if(char *dev, char *addr, char *reg,
...
+	if (err == -1 )
```
Comment (wkz) [inline phytool.c]:
```suggestion
	if (err == -1)
```

**Review by wkz — COMMENTED** [review]
After fixing the whitespace issues, please squash all commits on this PR into a single one (since it a single logical change) and force push it to this branch.

The summary line of that commit message should be something like "Handle errors from asprintf"

File: phytool.c (hunk 3 above)
Comment (smooge) [reply phytool.c]: I have merged past and fixed the lines above

**Pull request merged**